src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -348,6 +388,37 @@ private void EnsureSignature()
             }
         }
 
+        private void EnsureAccessorsBinding()
+        {
+            if (!_state.HasComplete(CompletionPart.FinishPropertyAccessorsBinding))
+            {
+                if (_state.NotePartComplete(CompletionPart.StartPropertyAccessorsBinding))
+                {
+                    // Ensure the binding is done so we guarantee that we have the BackingField set if necessary.
+                    // If we already have a backing field. We should still do binding since we create a backing field just if
+                    // we see an initializer. In this case binding is needed to know whether it's an auto property.
+                    if (this is SourcePropertySymbol { ContainsFieldKeyword: true } propertySymbol)
+                    {
+                        if (propertySymbol.GetMethod is SourceMemberMethodSymbol getMethod)
+                        {
+                            var binder = getMethod.TryGetBodyBinder();
+                            binder?.BindMethodBody(getMethod.SyntaxNode, BindingDiagnosticBag.Discarded);
+                        }
+                        if (propertySymbol.SetMethod is SourceMemberMethodSymbol setMethod)
```

Comment:
@AlekseyTs Yes. I'll see if checking for "IsAutoProperty" flag before binding is enough to avoid unnecessary binding.

Reply:
> I'll see if checking for "IsAutoProperty" flag before binding is enough to avoid unnecessary binding.

We want to know if the field isn't needed or has been created already. I would simply check the value of ```_lazyBackingFieldSymbol``` instead of relying on some other state that might or might not be related/equavalent.

Reply:
@AlekseyTs Accessor binding doesn't only create the backing field if necessary, but can change "IsAutoProperty" from false to true even if we already had a backing field. I already have a comment explaining that:

```csharp
            // We want to do this check regardless whether _lazyBackingFieldSymbol is null or not.
            // Consider `public int P { get => field; } = 0;
            // We'll create the backing field because we saw an initialzer, but we haven't yet determined
            // whether it's an auto property.
```


However, I think I should stop coding a bit and check couple of things:

- Do we really want `IsAutoProperty` to be true for `field` properties.
- What about `public string S { get; set => field = value; }` and `public string S { get => field; set; }`

This mostly depends on the areas where `IsAutoProperty` is referenced and what behavior do we want. If both behaviors are needed by different callers, we'll need to introduce some new API for that (eg `IsSemiAutoProperty`).

If it happened we don't want `IsAutoProperty` to be true for field, then `CreateBackingField` will not need to alter `_propertyFlags` after the symbol is created, and this may simplify things a lot.

@AlekseyTs Any strong opinion here about `IsAutoProperty` behavior?

Reply:
> Any "initial" strong opinion here about IsAutoProperty behavior?

My initial strong opinion is to not change behavior of ```IsAutoProperty``` unless we absolutely have to. First, we need to have a clear understanding why would we need a property with different behavior. That understanding shouldn't stop at recognizing that some test is failing. We should understand why it is failing, what portion of the compiler misbehaves and why.

Reply:
Yeah I see `IsAutoProperty` behavior shouldn't change. I was mostly confused due to some other bad changes I made which are now reverted.

So back to the original question, `_lazyBackingFieldSymbol` is not sufficient. The test `TestSemiAutoPropertyWithInitializer` will fail if we did so. The binding doesn't only create the field, but also sets the `IsAutoProperty` flag.